### PR TITLE
Add ability to auto-generate alembic migrations

### PR DESCRIPTION
**Pull request by portante**
We move to a slightly more general model where we can support specific Alembic operations passed as arguments.

This was very useful to easily run the auto-generated migrations.

### CODE REVIEW

**Review by dbutenhof — APPROVED** [review]

**Review by webbnh — CHANGES_REQUESTED** [review]
After this morning's discussions, I would like to see this rebased on ~#3203, which is now ready to merge pending one more approval~ `main`.

**Review by portante — COMMENTED** [review]
Rebased on #3211 and now depends on that PR.

**Review by webbnh — APPROVED** [review]
👍 

(I recognize that this PR is in draft mode, but I'm trying to stay ahead of the curve, here.)

**Review by portante — COMMENTED** [review]
Ready for review.

**Review by dbutenhof — APPROVED** [review]

**Pull request merged**
